tests/regression/tools/notification/base_client.c

```diff
@@ -0,0 +1,404 @@
+	if (condition_status != LTTNG_CONDITION_STATUS_OK) {
+		printf("error: Could not set domain type\n");
+		ret = 1;
+		goto end;
+	}
+
+	action = lttng_action_notify_create();
+	if (!action) {
+		printf("error: Could not create action notify\n");
+		ret = 1;
+		goto end;
+	}
+
+	trigger = lttng_trigger_create(condition, action);
+	if (!trigger) {
+		printf("error: Could not create trigger\n");
+		ret = 1;
+		goto end;
+	}
+
+	ret = lttng_register_trigger(trigger);
+
+	/*
+	 * An equivalent trigger might already be registered if an other app
+	 * registered one with similar action and condition
+	 */
+	if (ret < 0 && ret != -LTTNG_ERR_TRIGGER_EXISTS) {
+		printf("error: %s\n", lttng_strerror(ret));
+		ret = 1;
+		goto end;
+	}
+
+	nc_status = lttng_notification_channel_subscribe(notification_channel, condition);
+	if (nc_status != LTTNG_NOTIFICATION_CHANNEL_STATUS_OK) {
+		printf("error: Could not subscribe\n");
+		ret = 1;
+		goto end;
+	}
+
+	/* Print to stdout for readyness */
+	printf("sync: ready\n");
+
+	for (;;) {
+		struct lttng_notification *notification;
+		enum lttng_notification_channel_status status;
+		const struct lttng_evaluation *notification_evaluation;
+		const struct lttng_condition *notification_condition;
+
+		if (nr_notifications == nr_expected_notifications) {
+			ret = 0;
+			goto end;
+		}
+		/* Receive the next notification. */
+		status = lttng_notification_channel_get_next_notification(
+				notification_channel,
+				&notification);
+
+		switch (status) {
+		case LTTNG_NOTIFICATION_CHANNEL_STATUS_OK:
+			break;
+		case LTTNG_NOTIFICATION_CHANNEL_STATUS_NOTIFICATIONS_DROPPED:
+			ret = 1;
+			printf("error: No drop should be observed during this test app");
+			goto end;
+		case LTTNG_NOTIFICATION_CHANNEL_STATUS_CLOSED:
+			/*
+			 * The notification channel has been closed by the
+			 * session daemon. This is typically caused by a session
+			 * daemon shutting down (cleanly or because of a crash).
+			 */
+			printf("error: Notification channel was closed");
+			ret = 1;
+			goto end;
+		default:
+			/* Unhandled conditions / errors. */
+			ret = 1;
+			goto end;
+		}
+
+		notification_condition = lttng_notification_get_condition(notification);
+		notification_evaluation = lttng_notification_get_evaluation(notification);
+
+		ret = handle_condition(notification_condition, notification_evaluation);
+		nr_notifications++;
+
+		lttng_notification_destroy(notification);
+		if (ret != 0) {
+			goto end;
+		}
+	}
+end:
+	if (trigger) {
+		lttng_unregister_trigger(trigger);
+	}
+	lttng_notification_channel_unsubscribe(notification_channel, condition);
+	lttng_trigger_destroy(trigger);
+	lttng_condition_destroy(condition);
+	lttng_action_destroy(action);
+	lttng_notification_channel_destroy(notification_channel);
+	printf("exit: %d", ret);
+	return ret;
+}
+
+int handle_condition(
+		const struct lttng_condition *condition,
+		const struct lttng_evaluation *evaluation)
+{
+	int ret = 0;
+
+	/* Printing var */
```

Comment:
This can be removed. It state that the following vars are used for printing/synchronization.